src/formatter.ts

```diff
@@ -560,12 +575,60 @@ export class Formatter {
   }
 
   /** Create a `ModifierContext` for each tick in `voices`. */
-  createModifierContexts(voices: Voice[]): AlignmentContexts<ModifierContext> {
-    const fn: addToContextFn<ModifierContext> = (tickable: Tickable, context: ModifierContext) =>
-      tickable.addToModifierContext(context);
-    const contexts = createContexts(voices, () => new ModifierContext(), fn);
-    this.modifierContexts = contexts;
-    return contexts;
+  createModifierContexts(voices: Voice[]) {
```

Comment:
What's different here from `createContexts`? Is there no way to share the code?

Reply:
The difference is required by Systems with several Staves. In these cases:
- There is only one TickContext for each tick. This TickContext is shared by all the staves.
- There are N ModifierContext for each tick, where N is the number of staves. ModiferContexts are not shared between Staves.

This is the reason why `AlignmentModifierContexts` below is defining map keyed by Stave. 

I see not easy way to share the code.

Reply:
sounds good, thanks